operator/v1alpha1/types.go

```diff
@@ -109,3 +111,33 @@ type OperatorStatus struct {
 	// targetVersionAvailability is availability information for the target version if we are migrating
 	TargetAvailability *VersionAvailablity `json:"targetVersionAvailability,omitempty" protobuf:"bytes,6,opt,name=targetVersionAvailability"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GenericOperatorConfig provides information to configure an operator
+type GenericOperatorConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// ServingInfo is the HTTP serving information for the controller's endpoints
+	ServingInfo configv1.HTTPServingInfo `json:"servingInfo,omitempty" protobuf:"bytes,4,opt,name=servingInfo"`
+
+	// leaderElection provides information to elect a leader. Only override this if you have a specific need
+	LeaderElection configv1.LeaderElection `json:"leaderElection,omitempty" protobuf:"bytes,5,opt,name=leaderElection"`
+
+	// authentication allows configuration of authentication for the endpoints
+	Authentication DelegatedAuthentication `json:"authentication,omitempty" protobuf:"bytes,2,opt,name=authentication"`
+	// authorization allows configuration of authentication for the endpoints
+	Authorization DelegatedAuthorization `json:"authorization,omitempty" protobuf:"bytes,3,opt,name=authorization"`
```

Comment:
protobuf tags are inconsistent

Reply:
> protobuf tags are inconsistent

inconsistent with what?

Reply:
> protobuf tags are inconsistent

Oh, the order.  It doesn't really matter.  We aren't going to support using proto to encode you're config, but will update.